roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -96,43 +97,69 @@ void RooMinimizer::cleanup()
 /// for HESSE and MINOS error analysis is taken from the defaultErrorLevel()
 /// value of the input function.
 
-RooMinimizer::RooMinimizer(RooAbsReal &function, FcnMode fcnMode) : _fcnMode(fcnMode)
+/// Construcor that accepts all configuration in struct with RooAbsReal likelihood
+RooMinimizer::RooMinimizer(RooAbsReal &function, Config const &cfg) : _cfg(cfg)
 {
+   // Call init functions
    initMinimizerFirstPart();
+   if (_cfg.parallelGradient || _cfg.parallelLikelihood) {
+      auto nll_real = dynamic_cast<RooFit::TestStatistics::RooRealL *>(&function);
+      if (nll_real == nullptr)
+         throw std::logic_error("In RooMinimizer constructor: Selected parallel gradient or likelihood evaluation "
+                                "but an old-style likelihood was given. Please supply NewStyle(true) as an "
+                                "argument to createNLL for new-style likelihoods.");
+      initMultiProcess(nll_real->getRooAbsL());
+   } else {
+      initSerial(function);
+   }
```

Comment:
See also what I wrote on Mattermost to you, but I would not expect the minimizer to error out when passing `parallelGradient == true`, `parallelLikelihood == false` and having an old test statistic object.

In this case, I would expect that the code branch with the `RooGradMinimizerFcn` is taken, which works with the old test statistics too. But that makes the `fcnMode` redundant then, right?